### PR TITLE
Fix section depth on layout view for draft casebooks

### PR DESCRIPTION
**Pull request by abziegler**
This PR attempts to address #478. I'm a bit out of my comfort zone on this one @cgruppioni, so you may want to give this one a very close look when you can. It does work locally for me.

- Rewrote the layout/edit view from HAML to ERB
- Added the section depth logic from the layout/show view
- Modified the main layout template to pull in the right data (I think)

Note: there is one bit of code in the old view that I couldn't make sense of. On Line 2 of the old view there was `- content = edit`. I did not include that in the ERB and cannot tell if what it is doing.

### CODE REVIEW

File: app/views/content/table_of_contents/_edit.erb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            <button class="action-delete"></button>
+          </div>
+        <% end %>
+      </div>
+    <% end %>
+  <% elsif content.is_a? Enumerable %>
+      <div class="section-wrapper">
+      <% if editable %>
+        <div class="section-contents editable">
+          <%= render 'content/table_of_contents/edit', collection=content %>
```
Comment (cgruppioni) [inline app/views/content/table_of_contents/_edit.erb]: collection: content ( and on line 73)

File: app/views/content/table_of_contents/_edit.erb [hunk 2]
```diff
@@ -0,0 +1,84 @@
+<% editable = true %>
+<% start_depth = contents.first.try(:ordinals).try(:length) %>
+<% current_depth = start_depth %>
+
+<% contents.each do |content| %>
+  <%# render separate opening and closing tags around sections %>
+  <% (1..content.ordinals.length - current_depth).each do %>
```
Comment (cgruppioni) [inline app/views/content/table_of_contents/_edit.erb]: I pulled the code down to try to figure this out more. If it's the first resource in a section insert the <div> tags on top of it?

**Comment by cgruppioni** [comment]
Do you know why this was only happening for sections and not for casebooks?

**Comment by abziegler** [comment]
@cgruppioni I don't know why. Good question. 

What's happening is, in draft mode, sections are progressively nesting / increasing their depth, except when the full casebook contents are displayed.

**Pull request merged**